packages/components/src/button/test/index.js

```diff
@@ -12,172 +13,197 @@ import { plusCircle } from '@wordpress/icons';
+				.firstChild;
+
+			expect( button ).toHaveClass( 'has-icon' );
+			expect( button ).not.toHaveAttribute( 'aria-label' );
 		} );
 
 		it( 'should render a Dashicon component matching the wordpress icon', () => {
-			const iconButton = shallow( <Button icon={ plusCircle } /> ).find(
-				'button'
+			const button = render( <Button icon={ plusCircle } /> ).container
+				.firstChild;
+
+			expect( button ).toContainElement(
+				screen.getByTestId( 'test-icon' )
 			);
-			expect( iconButton.find( 'Icon' ).dive() ).not.toBeNull();
 		} );
 
 		it( 'should render child elements and icon', () => {
-			const iconButton = shallow(
+			const button = render(
 				<Button
 					icon={ plusCircle }
 					children={ <p className="test">Test</p> }
 				/>
-			).find( 'button' );
-			expect( iconButton.find( 'Icon' ).dive() ).not.toBeNull();
-			expect( iconButton.find( '.test' ).shallow().text() ).toBe(
-				'Test'
+			).container.firstChild;
+
+			expect( button ).toContainElement(
+				screen.getByTestId( 'test-icon' )
 			);
+
+			const paragraph = button.childNodes[ 1 ];
+			expect( paragraph ).toHaveClass( 'test' );
+			expect( paragraph ).toHaveTextContent( 'Test' );
 		} );
 
 		it( 'should add an aria-label when the label property is used, with Tooltip wrapper', () => {
-			const iconButton = shallow(
-				<Button icon={ plusCircle } label="WordPress" />
-			).find( 'Tooltip' );
-			expect( iconButton.name() ).toBe( 'Tooltip' );
-			expect( iconButton.prop( 'text' ) ).toBe( 'WordPress' );
-			expect( iconButton.find( 'button' ).prop( 'aria-label' ) ).toBe(
-				'WordPress'
-			);
+			render( <Button icon={ plusCircle } label="WordPress" /> );
+
+			const tooltip = screen.getByTestId( 'test-tooltip' );
+			expect( tooltip ).toBeVisible();
+			expect( tooltip ).toHaveAttribute( 'title', 'WordPress' );
+
+			const button = screen.getByRole( 'button' );
+			expect( tooltip ).toContainElement( button );
+			expect( button ).toHaveAttribute( 'aria-label', 'WordPress' );
 		} );
 
 		it( 'should support explicit aria-label override', () => {
-			const iconButton = shallow( <Button aria-label="Custom" /> ).find(
-				'button'
-			);
-			expect( iconButton.prop( 'aria-label' ) ).toBe( 'Custom' );
+			const button = render( <Button aria-label="Custom" /> ).container
+				.firstChild;
+
+			expect( button ).toHaveAttribute( 'aria-label', 'Custom' );
 		} );
 
 		it( 'should support adding aria-describedby text', () => {
-			const buttonDescription = shallow(
-				<Button describedBy="Description text" />
-			)
-				.find( VisuallyHidden )
-				.shallow()
-				.text();
-			expect( buttonDescription ).toBe( 'Description text' );
+			const button = render( <Button describedBy="Description text" /> )
+				.container.firstChild;
+
+			expect( button ).toHaveAttribute( 'aria-describedby' );
+			expect(
+				screen.getByTestId( 'test-visually-hidden' )
+			).toHaveTextContent( 'Description text' );
```

Comment:
Something like this would be more robust, and eliminate the need for a VisuallyHidden mock:

```suggestion
			render( <Button describedBy="Description text" /> );
			const button = screen.getByRole( 'button', {
				description: 'Description text',
			} );

			expect( button ).toBeInTheDocument();
```

Reply:
This is great, thank you 👍

Reply:
Updated to use your suggestion here - it's a really great example why mocks are not recommended.